pyomo/contrib/interior_point/linalg/results.py

```diff
@@ -0,0 +1,14 @@
+import enum
+
+
+class LinearSolverStatus(enum.Enum):
+    successful = 0
+    not_enough_memory = 1
+    singular = 2
+    error = 3
+    warning = 4
+
```

Comment:
Do we call out not_enough_memory and singular because these are the only errors we do anything special with?

Reply:
Exactly. We can add more to this enum as needed.